onyo/commands/edit.py

```diff
@@ -4,40 +4,43 @@
 import os
 import sys
 
-from git import Repo
+from pathlib import Path
 
 from onyo.utils import (
     build_git_add_cmd,
-    run_cmd,
-    edit_file
+    edit_file,
+    get_editor,
+    get_list_of_assets,
+    run_cmd
 )
 
 logging.basicConfig()
 logger = logging.getLogger('onyo')
 
 
 def build_commit_cmd(files, onyo_root):
-    return ["git -C \"" + onyo_root + "\" commit -m", "edit asset(s).\n\n" + "\n".join(files)]
+    return [f'git -C "{onyo_root}" commit -m "edit asset(s).\n\n"', "\n".join(files)]
 
 
 def prepare_arguments(sources, onyo_root):
     problem_str = ""
     list_of_sources = []
+    asset_list = get_list_of_assets(onyo_root)
+
     if isinstance(sources, str):
         sources = ["".join(sources)]
     for source in sources:
-        current_source = source
-        if not os.path.isfile(current_source):
-            current_source = os.path.join(onyo_root, source)
-        if not os.path.exists(current_source):
-            problem_str = problem_str + "\n" + current_source + " does not exist."
+        current_source = Path(source)
+        if not current_source.is_file():
+            current_source = Path(onyo_root, source)
+        if not current_source.exists():
+            problem_str += f"\n{current_source} does not exist."
         # check if file is in git
-        run_output = run_cmd("git -C \"" + onyo_root + "\" ls-tree -r HEAD ")
-        if source not in run_output:
-            problem_str = problem_str + "\n" + current_source + " is not in onyo."
+        elif True not in [source in asset[0] for asset in asset_list]:
+            problem_str += f"\n{current_source} is not in onyo."
```

Comment:
This error message (and the supporting comments) could be more clear here.

If I understand it correctly, this is trying to guard against the situation where a user attempts to edit an untracked file, which would have to be in `.gitignore` (in order to pass `onyo fsck`). Yes?

Or... I see that you're using `asset_list`, which /currently/ returns a list of all files, but eventually will in fact only return files which match the asset pattern. Are you guarding against `onyo edit` being used on non-assets?

In either case, better messaging and comments would help understand the programmer's intent here.